pkg/apis/serving/v1alpha1/build_compat_test.go

```diff
@@ -0,0 +1,358 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package v1alpha1
+
+import (
+	"encoding/json"
+	"errors"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	buildv1alpha1 "github.com/knative/build/pkg/apis/build/v1alpha1"
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+)
+
+type Enclosing struct {
+	Build *RawExtension `json:"build,omitempty"`
+}
+
+func TestMarshal(t *testing.T) {
+	tests := []struct {
+		name string
+		obj  interface{}
+		want string
+	}{{
+		name: "omit raw extension",
+		obj:  &Enclosing{},
+		want: `{}`,
+	}, {
+		name: "empty raw extension",
+		obj: &Enclosing{
+			Build: &RawExtension{},
+		},
+		want: `{"build":null}`,
+	}, {
+		name: "raw extension with bytes",
+		obj: &Enclosing{
+			Build: &RawExtension{
+				Raw: []byte(`"this is a string"`),
+			},
+		},
+		want: `{"build":"this is a string"}`,
+	}, {
+		name: "raw extension with object",
+		obj: &Enclosing{
+			Build: &RawExtension{
+				Object: &buildv1alpha1.Build{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "build.knative.dev",
+						Kind:       "Build",
+					},
+					Spec: buildv1alpha1.BuildSpec{
+						Steps: []corev1.Container{{
+							Image: "busybox",
+						}},
+					},
+				},
+			},
+		},
+		// TODO(mattmoor): This should not include Status.
+		want: `{"build":{"kind":"Build","apiVersion":"build.knative.dev","metadata":{"creationTimestamp":null},"spec":{"steps":[{"name":"","image":"busybox","resources":{}}],"timeout":"0s"},"status":{"creationTime":null,"startTime":null,"completionTime":null,"stepStates":null,"stepsCompleted":null}}}`,
```

Comment:
How would status be omitted if it's not "omitempty"?

Reply:
`omitempty` doesn't work with structs, it would have to be a pointer.